### ISSUE TITLE
Clipboard operations should use the selection buffer when possible, like Opera

### ISSUE THREAD

**Issue by elebow** [OtterBrowser/otter-browser#844]
Opera used the selection buffer in X11, as opposed to the clipboard buffer that Qt uses by default. The Opera behavior is not compliant with tradition such as described at http://www.jwz.org/doc/x-cut-and-paste.html (note that Qt uses the term *selection* in place of *primary*), but I prefer it and I think a lot of people are used to it.

The branch at https://github.com/OtterBrowser/otter-browser/compare/master...elebow:clipboard-selection uses the selection buffer by default.

A more complete solution should detect X11 at compile time, and maybe also provide a configuration option to the user.

**Comment by Emdek** [comment]
I guess that we could consider an option for that, something like *Interface/ClipboardMode*, but without changing defaults.
Lets wait a bit with that, maybe there will be additional opinions.

**Comment by komoku** [comment]
Makes sense for X11, but not for Windows (where it's not standard and users would be really surprised), so the user's system would definitely need to be taken into account if defaults are changed.

**Comment by ersi-dnd** [comment]
When there are options and toggles in right places, everybody should be satisfied. I copy-paste a lot and I am definitely in favour of more options in this area.

**Comment by ersi-dnd** [comment]
I just tried this in Opera and Seamonkey (both in Linux)

- Select something
- Move to a form field or textarea, e.g. address field
- middle-click pastes the selection

This shows that Primary works in Opera and Seamonkey. Otter pastes from Clipboard at middle-click in address field. Opera's and Seamonkey's way is better, but configurability is always best.

**Comment by ersi-dnd** [comment]
Looks like midclick to paste selection works in website textareas. So it's working to an extent, but of course it should be working in address field and every other field in Otter, like it does in Seamonkey, old Opera, etc.

It's been quite a revelation for me to find out that this feature works in more places in Linux than terminals.

**Comment by Chocimier** [comment]
@elebow About Copy*ToClipboardAction: are you sure text should be copied only to selection? Text you meant says that explicit copying should use clipboard. Opera fills both.

**Comment by Chocimier** [comment]
About middle-clicking on Linux in address field: there is an option
already to fix it: AddressField/PasteAndGoOnMiddleClick (set it to "No").
The default value is Windows-user-friendly, sadly.

**Comment by myier** [comment]
This is quite disturbing for me as well. Once something is in the clipboard, we cannot simply copy and paste something else with the regular X11 system. Maybe the worse part is that the middle click pastes something, but not what's selected. Why not use ^v for clipboard paste and button 3 for X11 paste?

**Comment by myier** [comment]
Please fix this, it's so hard to use otter on GNU/Linux systems without copy paste features!

**Comment by ersi-dnd** [comment]
+1

**Comment by beastie1** [comment]
+1

A must, definitely! How did I miss this ticket?! I was thinking about this two days ago and there it appears today in my inbox hehe.
So, two days ago I was trying to add a **Paste and Go** mouse gesture by middle-clicking on an empty tab, but it continuously failed. Then I realized the X11 clipboard didn't work as expected in Otter and I actually had to explicitly copy the URL/keyword with **Ctrl+C**.
